akka-docs/src/main/paradox/stream/stages-overview.md

```diff
@@ -719,6 +719,22 @@ Transform each element in the stream by calling a mapping function with it and p
 
 ---------------------------------------------------------------
 
+### inspect
+
+Similar to `map`, however does not modify the passed through element, the returned value is ignored.
+
+This operation is useful for performing inspecting the passed through element, usually by means of side-effecting operations (such as `println`, or emitting metrics), for each element without having to modify it.
+
+For logging signals (elements, completion, error) consider using the `log` stage instead,
+along with appropriate `ActorAttributes.createLogLevels`.
```

Comment:
If we are going to have this, I'd like a big clear warning here, and in the API-docs that it cannot touch mutable state etc. because this is exactly how people are going to misuse it.

Reply:
here you are: https://github.com/akka/akka/pull/24610/commits/577456f17c4cb8a4154f1cff4dd8e85574d641be